src/som/interpreter/actors/ReceivedMessage.java

```diff
@@ -35,9 +36,12 @@ public Object execute(final VirtualFrame frame) {
       dbg.prepareSteppingAfterNextRootNode();
     }
 
-    Object result = onReceive.doPreEvaluated(frame, msg.args);
-
-    resolvePromise(frame, msg.resolver, result, msg.triggerPromiseResolutionBreakpoint);
+    try {
+      Object result = onReceive.doPreEvaluated(frame, msg.args);
+      resolvePromise(frame, msg.resolver, result, msg.triggerPromiseResolutionBreakpoint);
+    } catch (SomException exception) {
+      msg.getResolver().onError(exception);
```

Comment:
Also, using `onError` directly here seems incorrect. Looking at the implementation of `onError()`, it is definitely incorrect.

The error case needs to be resolved very similarly to the normal case.
So, I'd expect the same code structure, using something like `resolvePromise(...)` from the normal case, too.
`resolvePromise()` uses the `ResolvePromiseNode` which does synchronization and handles when promises are already resolved, etc. All this, needs also to be done for the error case.